plugins/woocommerce-admin/client/marketplace/components/content/content.tsx

```diff
@@ -40,169 +42,157 @@ export default function Content(): JSX.Element {
+				setSearchResultsCount( {
+					extensions: combinedProducts.filter(
+						( p ) => p.type === ProductType.extension
+					).length,
+					themes: combinedProducts.filter(
+						( p ) => p.type === ProductType.theme
+					).length,
+					'business-services': combinedProducts.filter(
+						( p ) => p.type === ProductType.businessService
+					).length,
 				} );
-				return () => {
-					abortControllers.forEach( ( controller ) => {
-						controller.abort();
-					} );
-				};
-			}
-		);
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-	}, [] );
-
-	// Get the content for this screen
-	useEffect( () => {
-		const abortController = new AbortController();
-
-		if (
-			query.tab === undefined ||
-			( query.tab && [ '', 'discover' ].includes( query.tab ) )
-		) {
-			return;
-		}
-
-		setIsLoading( true );
-		setProducts( [] );
-
-		const params = new URLSearchParams();
-
-		if ( query.term ) {
-			params.append( 'term', query.term );
-		}
-
-		if ( query.category ) {
-			params.append(
-				'category',
-				query.category === '_all' ? '' : query.category
-			);
-		} else if ( query?.tab === 'themes' ) {
-			params.append( 'category', 'themes' );
-		} else if ( query?.tab === 'business-services' ) {
-			params.append( 'category', 'business-services' );
-		} else if ( query?.tab === 'search' ) {
-			params.append( 'category', 'extensions-themes-business-services' );
-		}
-
-		const wccomSettings = getAdminSetting( 'wccomHelper', false );
-		if ( wccomSettings.storeCountry ) {
-			params.append( 'country', wccomSettings.storeCountry );
-		}
-
-		fetchSearchResults( params, abortController.signal )
-			.then( ( productList ) => {
-				setProducts( productList );
-
-				if ( query.term ) {
-					setSearchResultsCount( {
-						extensions: productList.filter(
-							( p ) => p.type === 'extension'
-						).length,
-						themes: productList.filter(
-							( p ) => p.type === 'theme'
-						).length,
-						'business-services': productList.filter(
-							( p ) => p.type === 'business-service'
-						).length,
-					} );
-				}
 			} )
 			.catch( () => {
-				setProducts( [] );
+				setAllProducts( [] );
 			} )
 			.finally( () => {
-				// we are recording both the new and legacy events here for now
-				// they're separate methods to make it easier to remove the legacy one later
-				const marketplaceViewProps = {
-					view: query?.tab,
-					search_term: query?.term,
-					product_type: query?.section,
-					category: query?.category,
-				};
-
-				recordMarketplaceView( marketplaceViewProps );
-				recordLegacyTabView( marketplaceViewProps );
 				setIsLoading( false );
 			} );
+
 		return () => {
-			abortController.abort();
+			abortControllers.forEach( ( controller ) => {
+				controller.abort();
+			} );
 		};
-	}, [
-		query.term,
-		query.category,
-		query?.tab,
-		setIsLoading,
-		query?.section,
-	] );
+	}, [ query.term, query.category ] ); // Depend on term and category
```

Comment:
I'm getting a missing dependencies error for this line: 'setHasBusinessServices', 'setIsLoading', and 'setSearchResultsCount'

Reply:
These are already present [here](https://github.com/woocommerce/woocommerce/blob/9c18ebd77fc46491e3bbc53400fc86900ce3bc39/plugins/woocommerce-admin/client/marketplace/components/content/content.tsx#L38-L40)

Reply:
They're used within the useEffect hook, but not included in the dependency array for that hook.

Reply:
Hmm, what gives the "missing dependencies error"? Is it a common practice to include functions (that do not change) in a dependency array?

Reply:
I'm getting the error in my IDE:

![content_tsx_—_woocommerce](https://github.com/user-attachments/assets/4b70a6d6-6a60-4848-9eb6-4de6b7a632ee)

Here's the [React docs](https://react.dev/reference/react/useEffect#specifying-reactive-dependencies) on the topic.

> Is it a common practice to include functions (that do not change)

The function doesn't change, but the value it returns can change, if my understanding is correct. Cursor suggests memoizing the functions with a useCallback hook if you don't want to trigger unnecessary re-renders.

Reply:
Gotcha! Thanks. Fixed in 0b2d2dca6d